odxtools/odxlink.py

```diff
@@ -359,6 +372,12 @@ def resolve_snref(target_short_name: str,
                  f"object while expecting {expected_type.__name__}")
 
     if use_weakrefs:
+        if isinstance(candidates[0], (weakref.CallableProxyType, weakref.ProxyType)):
+            # if the referenced object already is a
+            # weakref proxy object, we can and must return
+            # it directly. (weakref.proxy() raises an
+            # exception if called for a proxy object.)
+            return candidates[0]
```

Comment:
make a wrapper/helper function instead of repeating same logic 3 times

Reply:
To avoid repeating the comment/doc as well

Reply:
my personal rule of thumb with duplication is that if the block has more than 3 lines (comments included), or repeats more than three times, than it needs to go to its own function.

the "block" in this case is not just the proxy creation, but extends to the `use_weakrefs` check, since it's also duplicated